core/eolearn/core/eodata_io.py

```diff
@@ -403,12 +430,12 @@ def _jsonify_timestamp(param: object) -> str:
     raise TypeError(f"Object of type {type(param)} is not yet supported in jsonify utility function")
 
 
-def _create_feature_io(ftype: FeatureType, path: str, filesystem: FS) -> FeatureIO:
+def _get_feature_io_constructor(ftype: FeatureType) -> Type[FeatureIO]:
     """Creates the correct FeatureIO, corresponding to the FeatureType."""
     if ftype is FeatureType.BBOX:
-        return FeatureIOBBox(path, filesystem)
+        return FeatureIOBBox
     if ftype in (FeatureType.TIMESTAMP, FeatureType.META_INFO):
-        return FeatureIOJson(path, filesystem)
+        return FeatureIOJson
     if ftype in FeatureTypeSet.VECTOR_TYPES:
-        return FeatureIOGeoDf(path, filesystem)
-    return FeatureIONumpy(path, filesystem)
+        return FeatureIOGeoDf
+    return FeatureIONumpy
```

Comment:
shouldn't there be another if and default return value some error flag ?

Reply:
This would occur only if we would add a new feature type that requires a new FeatureIO object, and forget to handle it here (but there would be errors as soon as one would try to save it), so I don't think it's worth another 3 lines of code :thinking: